R/Rules-class.R

```diff
@@ -1037,6 +1037,78 @@ StoppingMTDCV <- function(target = 0.3,
   )
 }
 
+
+# StoppingLowestDoseHSRBeta-class ----
+
+#' `StoppingLowestDoseHSRBeta`
+#'
+#' @description `r lifecycle::badge("experimental")`
+#'
+#' [`StoppingLowestDoseHSRBeta`] is a class for stopping based on a Hard Safety
+#' Rule using the Beta posterior distribution with Beta(a,b) prior and a
+#' Bin-Beta model based on the observed data at the lowest dose level.
+#' The rule is triggered when the first dose is considered to be toxic
+#' (i.e. above threshold probability) based on the observed data at the
+#' lowest dose level and a Beta(a,b) prior distribution.
+#' The default prior is Beta(1,1).
+#' In case that placebo is used, the rule is evaluated at the second dose of the
+#' dose grid, i.e. at the lowest non-placebo dose.
+#' Note: this stopping rule is independent from the underlying model.
+#'
+#' @slot target (`number`)\cr the target toxicity.
+#' @slot prob (`number`)\cr the threshold probability for the lowest
+#'  dose being toxic.
```

Comment:
```suggestion
#' @slot target (`proportion`)\cr the target toxicity.
#' @slot prob (`proportion`)\cr the threshold probability for the lowest
#'  dose being toxic.
```
even though the R class is numeric we relate to this as proportion here for the user and in the validation function